sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarReservoir.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.metrics.data.Exemplar;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An interface for an exemplar reservoir of samples.
+ *
+ * <p>This represents a reservoir for a specific "point" of metric data.
+ */
+public interface ExemplarReservoir {
```

Comment:
Reservoir usually implies a type of sampling, should this be the class that's called `ExemplarSampler`? It looks very similar to the tracing Sampler

Reply:
the reservoir is instantiated *per* metric-timeseries.  I.e. one ExemplarSampler will be instantiating many reservoirs.

I tried to keep the two as similar as possible, only deviating where it makes sense, but I could have gotten the split wrong.